qabelbox/src/test/java/de/qabel/qabelbox/identity/IdentityDetailsPresenterTest.kt

```diff
@@ -0,0 +1,81 @@
+package de.qabel.qabelbox.identity
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.stub
+import com.nhaarman.mockito_kotlin.verify
+import de.qabel.core.extensions.CoreTestCase
+import de.qabel.core.extensions.createIdentity
+import de.qabel.qabelbox.BuildConfig
+import de.qabel.qabelbox.SimpleApplication
+import de.qabel.qabelbox.eq
+import de.qabel.qabelbox.identity.interactor.IdentityUseCase
+import de.qabel.qabelbox.identity.view.IdentityDetailsView
+import de.qabel.qabelbox.identity.view.presenter.IdentityDetailsPresenter
+import de.qabel.qabelbox.identity.view.presenter.MainIdentityDetailsPresenter
+import de.qabel.qabelbox.navigation.Navigator
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricGradleTestRunner
+import org.robolectric.annotation.Config
+import rx.lang.kotlin.singleOf
+import rx.lang.kotlin.toSingletonObservable
+
+@RunWith(RobolectricGradleTestRunner::class)
+@Config(application = SimpleApplication::class, constants = BuildConfig::class)
+class IdentityDetailsPresenterTest() : CoreTestCase {
+
+    val identity = createIdentity("Bob")
+
+    val view: IdentityDetailsView = mock<IdentityDetailsView>().apply {
+        stub(identityKeyId).toReturn(identity.keyIdentifier)
+    }
+    val useCase: IdentityUseCase = mock<IdentityUseCase>().apply {
+        stub(getIdentity(identity.keyIdentifier)).toReturn(identity.toSingletonObservable().toSingle())
+        stub(saveIdentity(identity)).toReturn(singleOf(Unit))
+    }
+
+    val navigator: Navigator = mock()
+
+    val presenter: IdentityDetailsPresenter = MainIdentityDetailsPresenter(view, useCase, navigator)
+
+    @Test
+    fun testLoad() {
+        presenter.loadIdentity()
+        verify(view).loadIdentity(identity)
+        presenter.identity eq identity
+    }
+
+    @Test
+    fun testSaveAlias() {
+        presenter.identity = identity
+        presenter.onSaveAlias("Alice")
+        verify(view).loadIdentity(identity)
+        identity.alias eq "Alice"
+    }
+
+    @Test
+    @Ignore("TODO Not working in roboelectic")
```

Comment:
If there is no way to run it with robolectric (like by exchanging the numberformatter or whatever robolectric doesn't like here by some testable implementation, even if it's just for the test), then you could maybe move this test to another class that doesn't use the robolectric runner.